nevergrad/optimization/optimizerlib.py

```diff
@@ -1869,6 +1869,68 @@ def __init__(self, parametrization: IntOrParameter, budget: tp.Optional[int] = N
         optim = self.optim if not isinstance(self.optim, NGO) else self.optim.optim
         logger.debug("%s selected %s optimizer.", *(x.name for x in (self, optim)))
 
+@registry.register
+class ShiwaC1(Shiwa):
+    def __init__(self, parametrization: IntOrParameter, budget: tp.Optional[int] = None, num_workers: int = 1) -> None:
+        super().__init__(parametrization, budget=budget, num_workers=num_workers) 
+        self.set_constraints_management(max_constraints_trials = 1000, constraint_penalization = 1., constraint_penalty_exponent = 1.0001,
+            penalize_cheap_violations = True, memorize_constraint_failures = False)
```

Comment:
Please remove all this variants, this is a lot of duplicated code and would make maintenance too complicated.
You should create a ConstrainedNGOpt ConfiguredOptimizer, and then instantiate the ones you need in the experiments. No need to register them because they are too specific for being defined here.

Reply:
As mentioned in the comments, this is exactly the intent, this is here just for discussion until experiments conclude the debate.

Reply:
Then it should be in a separate PR that is not merged, anything merged is fair game and should be considered as a breaking changes if removed. So I definitely don't want to add something that will be removed right after.